arangod/Aql/MaterializeExecutor.h

```diff
@@ -47,11 +48,36 @@ class RegisterInfos;
 template<BlockPassthrough>
 class SingleRowFetcher;
 
+// two storage varians as we need
+// collection name to be sotred only
+// when needed.
+class NoCollectionNameHolder {};
+class StringCollectionNameHolder {
+ public:
+  StringCollectionNameHolder(std::string const& name)
+      : _collectionSource(name) {}
+
+ protected:
+  std::string const& _collectionSource;
```

Comment:
Maybe string_view better?

Reply:
We anyway have a string reference here (it comes from the collection"name() call). And there is no way to have anything else. So let it be a reference for now.

Reply:
Ok